ToolBox/TradingEconomicsDataDownloader/TradingEconomicsCalendarDownloader.cs

```diff
@@ -94,13 +97,27 @@ public override bool Run()
                     _requestGate.WaitToProceed(TimeSpan.FromSeconds(1));
 
                     var content = Get(startUtc, endUtc).Result;
-                    var collection = JsonConvert.DeserializeObject<List<TradingEconomicsCalendar>>(content);
+                    var rawCollection = JsonConvert.DeserializeObject<JArray>(content);
+
+                    foreach (var rawData in rawCollection)
+                    {
+                        var inPercentage = rawData["Actual"].Value<string>().Contains("%");
+
+                        rawData["IsPercentage"] = inPercentage;
+                        rawData["Actual"] = ParseDecimal(rawData["Actual"].Value<string>(), inPercentage);
+                        rawData["Previous"] = ParseDecimal(rawData["Previous"].Value<string>(), inPercentage);
+                        rawData["Forecast"] = ParseDecimal(rawData["Forecast"].Value<string>(), inPercentage);
+                        rawData["TEForecast"] = ParseDecimal(rawData["TEForecast"].Value<string>(), inPercentage);
+                        rawData["Revised"] = ParseDecimal(rawData["Revised"].Value<string>(), inPercentage);
+                    }
+
+                    var collection = rawCollection.ToObject<List<TradingEconomicsCalendar>>();
```

Comment:
Agree with @mchandschuh idea! If this were a custom JSON converter it would be beautiful ⭐️

Reply:
I played around with using a JSON Converter, but ultimately it didn't pan out, so this was the next best solution I could think of